File: fron-end/src/app/pedidos/pedido/pedido.ts
import { DetallePedido } from "./detallePedido";

export class Pedido{

  constructor(
    public pedidoId: number,
    public nro: string,
    public fecha: Date,
    public clienteId: number,
    public detallePedido: DetallePedido[]
  )
  {}

}

File: back-end/src/main/java/CRM4RTONOCTURNOA/CRM/entity/Pedido.java
package CRM4RTONOCTURNOA.CRM.entity;


import java.sql.Timestamp;
import java.util.HashSet;
import java.util.Set;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.MappedCollection;
import org.springframework.data.relational.core.mapping.Table;

import lombok.Data;

@Data
@Table("pedidos\".\"pedidos")
public class Pedido {

    @Id
    @Column("pedido_id")
    private long pedidoId;
    private String nro;
    private Timestamp fecha;
    @Column("cliente_id")
    private long clienteId;
    private String observacion;
    
    //Raiz
   @MappedCollection(idColumn = "pedido_id")
    Set<DetallePedido> detallePedido = new HashSet<>();
}
